sklearn/model_selection/_search.py

```diff
@@ -877,10 +877,13 @@ def _get_routed_params_for_fit(self, params):
         else:
             params = params.copy()
             groups = params.pop("groups", None)
+            score = {}
+            if "sample_weight" in params:
+                score["sample_weight"] = params["sample_weight"]
```

Comment:
We should not forward to scorers that do not accept sample weights and instead issue a warning.